spec.md

```diff
@@ -224,7 +224,9 @@ help intermediate gateways determine how to route the events.
 * Constraints:
   * REQUIRED
   * MUST be a non-empty string
-  * MUST be unique within the scope of the producer
+  * The producer MUST ensure that source+id is unique for each distinct event.
```

Comment:
I like the overall direction of this PR. Do we need to maintain the concept of the scope of the uniqueness? Meaning, is it globally unique vs just within the scope of the producer?

Reply:
I added some explanatory text/examples on scope of uniqueness for different forms of `source` URI reference. Does that cover it?

Reply:
```suggestion
  * The producer MUST ensure that source+id is unique for each distinct event within their scope.
```

Otherwise it reads as if the producer MUST have consumed all existing events to be sure that no one else has already produced an event with that source and id.

Reply:
I think that new text does help - thanks. And I agree with @cneijenhuis on using "event producer" instead of 'app". My last concern is placement of that text since it seems to apply to both `source` and `id` sections. Perhaps just say "The exact syntax of `source`, and the scope in which it (and `id`) is unique,...." to add a forward ref to make it clear the unique text applies to `id` as well.  Just a thought.

Reply:
It should be possible to ensure that the source+id combination is globally unique using either of the following techniques:

1. By using an authority in the source which is assigned to the source via an authoritative registry (e.g. DNS).
2. By using a GUID as a component of the eventID.

Reply:
resolving, please check latest and re-raise any issues.